internal/action/actions.go

```diff
@@ -160,6 +160,24 @@ func (h *BufPane) Center() bool {
 	return true
 }
 
+// CursorToView moves the cursor within the view if it's currently out of view.
+// The cursor is moved to an offset of 25% from the top of the view.
+func (h *BufPane) CursorToView() bool {
+	v := h.GetView()
+	height := h.BufView().Height
+	cursorSloc := h.SLocFromLoc(h.Cursor.Loc)
+	if cursorSloc.LessThan(v.StartLine) || cursorSloc.GreaterThan(h.Scroll(v.StartLine, height-1)) {
+		h.Cursor.Deselect(true)
+		h.Cursor.GotoLoc(h.LocFromVLoc(display.VLoc{
+			SLoc:    h.Scroll(v.StartLine, height/4),
+			VisualX: 0,
+		}))
+		h.RemoveAllMultiCursors()
```

Comment:
It probably makes more sense to remove multicursors before moving the cursor, not after it, since we want to move just one cursor? (Although the end result is probably the same.)

Probably even before deselect. See e.g. `GotoCmd()` and `JumpCmd()` in `internal/action/command.go`.

Reply:
That's the way I originally had it, but it causes strange behaviour if there is a multi-cursor. Try moving `h.RemoveAllMultiCursors()` up before `.Deselect()` and then try this:
- Open a file in Micro.
- Create a multi-cursor, turn it into a selection if you want.
- Move the multi-cursor out of view by using `PageDown` or `PageUp`.
- Invoke `CursorToView`.

Reply:
Hmm, you are right. That's because `RemoveAllMultiCursors()` calls `h.Relocate()` which changes the current view, so it gets in the way.

But what this means is that `CursorToView()` should probably directly call `h.Buf.ClearCursors()` (like `MousePress()` does, for instance) instead of calling `RemoveAllMultiCursors()`.